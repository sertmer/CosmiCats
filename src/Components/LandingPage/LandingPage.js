import React from 'react';
import './LandingPage.scss';
import { Link } from 'react-router-dom';


export const LandingPage = () => {
  return (
    <section>
      <h2>The Truth has Nine Lives</h2>
      <Link to={'/login'}>
        <button type='button' className='Login-button'>Join Us</button>
      </Link>
    </section>
  )
}

export default LandingPage;