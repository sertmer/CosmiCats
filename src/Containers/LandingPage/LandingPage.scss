@import '../../_scss/mixins';
@import '../../_scss/variables';

.landing-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  // justify-content: space-around;
  margin-top: 10%;
}

.tagline {
  font-size: 5em;
  color: $alien-green;
}

.login-button {
  width: 10em;
  // height: 8em;
  @include rounded-corners(5px);
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  padding: 3%;
  @include rounded-corners(5px);
  font-size: 2em;
  font-family: $main-font;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.login-button:hover {
  transform: scale(1.1);
}