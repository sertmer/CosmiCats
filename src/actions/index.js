export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setNasaImages = images => ({
  type: 'SET_NASA_IMAGES',
  images
})

export const setCatImages = images => ({
  type: 'SET_CAT_IMAGES',
  images
})

export const addToEvidence = evidence => ({
  type: 'ADD_EVIDENCE',
  evidence
})