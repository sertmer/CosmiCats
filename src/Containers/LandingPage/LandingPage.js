import React, { Component } from 'react';
import { getNasaImages, getCatsInSpace } from '../../utils/apiCalls';
import { setNasaImages, setCatImages } from '../../actions/';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './LandingPage.scss';

export class LandingPage extends Component {
  
  componentDidMount() {
    getNasaImages()
      .then(data => this.cleanData(data))
      .then(data => this.props.setNasaImages(data))
      .catch(err => console.log(err))
    this.createCatsArray()
  }

  createCatsArray = () => {
    let cats = []
    for (let i = 0; i < 20; i++) {
      getCatsInSpace()
        .then(data => cats.push(data[0]))
    }
    this.props.setCatImages(cats)
  }

  cleanData = (data) => {
    let cleanedData;
      return data.collection.items.map(item => {
        return cleanedData = {
          img: item.links[0].href,
          id: item.data[0].nasa_id,
          dateCreated: item.data[0].date_created
        }
      }
    )
  }

  render() {
    return (
      <section>
        <h2>The Truth has Nine Lives</h2>
        <Link to={'/login'}>
          <button type='button' className='Login-button'>Join Us</button>
        </Link>
      </section>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setNasaImages: images => dispatch(setNasaImages(images)),
  setCatImages: images => dispatch(setCatImages(images))
})

export default connect(null, mapDispatchToProps)(LandingPage);