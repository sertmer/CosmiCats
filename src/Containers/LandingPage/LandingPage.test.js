import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage, mapDispatchToProps } from './LandingPage';
import { setNasaImages } from '../../actions';

describe('LandingPage', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow(<LandingPage />)
  })

  describe('Unit Tests', () => {
    it.skip('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with images when setImages is invoked', () => {
      const mockDispatch = jest.fn();
      const mockActionToDispatch = setNasaImages([
        {
          img: "nasaimage.com",
          id: "ARC-1983-AC83-0565-1",
          dateCreated: "1983-08-01T00:00:00Z"
        }, 
        {
          img: "nasaimage.com",
          id: "ADC-1934-AC83-0565-3",
          dateCreated: "1977-09-02T00:00:00Z"
        }
      ])
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.setNasaImages([
        {
          img: "nasaimage.com",
          id: "ARC-1983-AC83-0565-1",
          dateCreated: "1983-08-01T00:00:00Z"
        }, 
        {
          img: "nasaimage.com",
          id: "ADC-1934-AC83-0565-3",
          dateCreated: "1977-09-02T00:00:00Z"
        }
      ])

      expect(mockDispatch).toHaveBeenCalledWith(mockActionToDispatch)
    })
  })
})