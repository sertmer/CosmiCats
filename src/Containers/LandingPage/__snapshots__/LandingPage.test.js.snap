// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Unit Tests should match the snapshot 1`] = `
<section>
  <h2>
    The Truth has Nine Lives
  </h2>
  <Link
    to="/login"
  >
    <button
      className="Login-button"
      type="button"
    >
      Join Us
    </button>
  </Link>
</section>
`;
